SDK/AppCenterData/Microsoft.AppCenter.Data.iOS/ConversionExtensions.cs

```diff
@@ -15,19 +15,30 @@ public static class ConversionExtensions
 
         public static DocumentWrapper<T> ToDocumentWrapper<T>(this MSDocumentWrapper documentWrapper)
         {
-            var jsonValue = documentWrapper.JsonValue;
-            var deserializedValue = jsonValue != null ? JsonConvert.DeserializeObject<T>(jsonValue) : default(T);
-            return new DocumentWrapper<T>
+            DocumentWrapper<T> parsedDocument = new DocumentWrapper<T>
             {
-                DeserializedValue = deserializedValue,
-                JsonValue = jsonValue,
                 Partition = documentWrapper.Partition,
                 ETag = documentWrapper.ETag,
                 Id = documentWrapper.DocumentId,
-                LastUpdatedDate = documentWrapper.LastUpdatedDate != null ? (DateTime)documentWrapper?.LastUpdatedDate : UnixEpoch,
                 IsFromDeviceCache = documentWrapper.FromDeviceCache
             };
-        }
+
+            try
+            {
+                var jsonValue = documentWrapper.JsonValue;
+                var deserializedValue = jsonValue != null ? JsonConvert.DeserializeObject<T>(jsonValue) : default(T);
+                parsedDocument.DeserializedValue = deserializedValue;
+                parsedDocument.JsonValue = jsonValue;
+                parsedDocument.LastUpdatedDate = documentWrapper.LastUpdatedDate != null ? (DateTime)documentWrapper?.LastUpdatedDate : UnixEpoch;
```

Comment:
I don't understand why the LastUpdatedDate has 2 variants here from try or catch.

The only exception that can happen here is newtonsoft json deserialization.